generator/src/test/java/com/scottlogic/deg/generator/generation/fieldvaluesources/datetime/SequentialDateIteratorTest.java

```diff
@@ -0,0 +1,124 @@
+package com.scottlogic.deg.generator.generation.fieldvaluesources.datetime;
+
+import org.junit.Assert;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+
+class SequentialDateIteratorTest {
+
+    private OffsetDateTime inclusiveMinDate;
+    private OffsetDateTime exclusiveMaxDate;
+    private OffsetDateTime referenceDate;
+
+    @BeforeEach
+    public void setup() {
+        // set some default values
+        this.inclusiveMinDate = OffsetDateTime.of(2001, 1, 1, 1, 1, 1, 1, ZoneOffset.UTC);
+        this.exclusiveMaxDate= OffsetDateTime.of(2010, 1, 1, 1, 1, 1, 1, ZoneOffset.UTC);
+    }
+
+    @Test
+    void HasNext_WhenCurrentIsEqualToMax_IsCorrect() {
```

Comment:
Small point about method naming, should start with lowercase e.g. `hasNext_whenCurrentIsEqualToMax_isCorrect`